File: core/src/main/java/com/minirogue/starwarscanontracker/core/model/room/entity/FilterObjectDto.java
package com.minirogue.starwarscanontracker.core.model.room.entity;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.ForeignKey;

/**
 * defines a filter for the media list
 */
@Entity(tableName = "filter_object",
        primaryKeys = {"filter_id", "type_id"},
        foreignKeys = @ForeignKey(entity = FilterTypeDto.class,
                parentColumns = "id",
                childColumns = "type_id",
                onDelete = ForeignKey.CASCADE))
public class FilterObjectDto {
    @ColumnInfo(name = "filter_id")
    public int id;
    @ColumnInfo(name = "type_id")
    public int filterType;
    @ColumnInfo(name = "is_active")
    public boolean active;
    @ColumnInfo(name = "filter_text")
    public String displayText;

    public FilterObjectDto(int id, int filterType, boolean active, String displayText) {
        this.id = id;
        this.filterType = filterType;
        this.active = active;
        this.displayText = displayText;
    }

    @NonNull
    @Override
    public String toString() {
        return ("id: " + id + " filterType: " + filterType + " active: " + active + " displayText" + displayText);
    }

    /**
     * Check if this FilterObject is equal to another object.
     * <p>
     * Returns True if they have the same id, column, and positive values,
     * but liveFilter and displayText are ignored
     *
     * @param obj the object being compared to this one
     */
    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj == null || obj.getClass() != this.getClass()) {
            return false;
        } else {
            FilterObjectDto otherObj = (FilterObjectDto) obj;
            return otherObj.id == this.id && otherObj.filterType == this.filterType;
        }
    }
}

File: core/src/commonMain/kotlin/com/holocanon/core/data/entity/FilterObjectDto.kt
package com.holocanon.core.data.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.ForeignKey.Companion.CASCADE

@Entity(
    tableName = "filter_object",
    primaryKeys = ["filter_id", "type_id"],
    foreignKeys = [
        ForeignKey(
            entity = FilterTypeDto::class,
            parentColumns = ["id"],
            childColumns = ["type_id"],
            onDelete = CASCADE,
        ),
    ],
)
data class FilterObjectDto(
    @ColumnInfo(name = "filter_id")
    val id: Int = 0,
    @ColumnInfo(name = "type_id")
    val filterType: Int = 0,
    @ColumnInfo(name = "is_active")
    val active: Boolean = false,
    @ColumnInfo(name = "filter_text")
    val displayText: String? = null,
)
